src/sentry/monitors/endpoints/organization_monitor_details.py

```diff
@@ -99,6 +99,7 @@ def put(self, request: Request, organization, project, monitor) -> Response:
         if "slug" in result:
             params["slug"] = result["slug"]
         if "status" in result:
+            # TODO(rjo100): resets status if monitor is failed, needs fixing with environments
             if result["status"] == MonitorStatus.ACTIVE:
```

Comment:
We're going to do a migration here to move all monitors that are in non `ObjectStatus` type status to ACTIVE or DISABLED right?

Reply:
ideally yes. we reference `PENDING DELETION` a lot too

Reply:
that's OK, pending deletion is part of ObjectStatus right